File: qc-core/src/main/java/com/eager/questioncloud/core/domain/hub/payment/repository/CouponRepository.java
package com.eager.questioncloud.core.domain.hub.payment.repository;

import com.eager.questioncloud.core.domain.hub.payment.model.Coupon;
import org.springframework.stereotype.Repository;

@Repository
public interface CouponRepository {
    Coupon findById(Long id);

    Coupon findByCode(String code);

    Coupon save(Coupon coupon);

    Boolean decreaseCount(Long couponId);
}

File: qc-core/qc-pay/qc-pay-core/src/main/kotlin/com/eager/questioncloud/payment/repository/CouponRepository.kt
package com.eager.questioncloud.payment.repository

import com.eager.questioncloud.payment.domain.Coupon

interface CouponRepository {
    fun findById(id: Long): Coupon
    
    fun findByIdIn(ids: List<Long>): List<Coupon>
    
    fun findByCode(code: String): Coupon
    
    fun save(coupon: Coupon): Coupon
    
    fun decreaseCount(couponId: Long): Boolean
    
    fun deleteAllInBatch()
}
